src/common/cluster-types.ts

```diff
@@ -19,6 +19,8 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+import Joi from "joi";
```

Comment:
Did you consider something like [Zod](https://github.com/colinhacks/zod) for more Typescript-first like validation library?

https://itnext.io/parse-dont-validate-incoming-data-in-typescript-d6d5bfb092c8

Reply:
I did not since we already have `joi` in the repo. Also we don't have strict mode on currently...

> ⚠️ IMPORTANT: You must enable strict mode in your tsconfig.json. This is a best practice for all TypeScript projects.

Yes I can see the benefit of the typings working as expected but I don't think it is worth it to introduce a new library just yet.

Reply:
Oh alright, I missed that `joi` is already in the project. Good points. Nevermind.